### PR TITLE
#2: Write tests

### LINKED ISSUE #2: Write Tests
## Story
As part of maturing the platform, we want a comprehensive test suite that exercises the public debt flows and guest onboarding paths so that regressions are caught early and we maintain confidence in every release.

## Acceptance Criteria
- The debt list flow can distinguish active debts from paid ones: after A settles, its payment is hidden but a new debt reappears, and the UI reflects that correctly.
- Unauthenticated visitors see the “Who are you?” prompt plus the “Maybe register?” cue on the account detail screen so the onboarding messaging stays consistent.
- Money flow summaries rendered in the frontend match the data already covered in backend tests, ensuring the templates consume the expected context.
- Guest users can register successfully, moving from the guest state into an authenticated account while maintaining their room context.
- The project achieves 100% test coverage across the relevant modules touched by these flows, and coverage reports are included with the PR.

## Technical Tasks
1. Turn the existing manual assertions into Django TestCases that drive the debt list view, verifying paid debts are hidden while new debts surface after a payment.
2. Add coverage-backed tests for the unauthenticated experience on account detail pages, asserting both onboarding prompts render.
3. Ensure the money flow templates are exercised by rendering the relevant views in tests and comparing their output to expected strings or context values.
4. Cover the guest user registration path, simulating the transition from guest to authenticated user and asserting post-login behavior.
5. Run coverage tooling (`pipenv run coverage run manage.py test` + `pipenv run coverage report`) and document the results so the branch proves 100% coverage for the targeted modules.

### PR DESCRIPTION
**Pull request by christoph-teichmeister**
/closes #2 

## Test Coverage for Public Debt Flows and Guest Onboarding

This PR implements the test suite specified in issue #2, covering critical user flows across debt management, guest onboarding, and account registration. The changes introduce 6 new test modules validating acceptance criteria and technical requirements.

### What's Being Tested

**Debt List Flow (2 test cases, 71 LOC)**
- Verifies outstanding debts render correctly with accurate counts and debtor/creditor associations
- Validates debt state transitions: settled debts disappear and new debts appear after settlement transactions are created
- Confirms the core requirement that the UI reflects active vs. paid debt status

**Guest Onboarding Experience (1 test case, 17 LOC)**
- Confirms unauthenticated users see "Who are you?" prompt plus "Maybe register?" registration cue on room dashboard
- Validates registration suggestion messaging ("Create a free account") renders to guests

**Money-Flow Aggregation (2 test cases)**
- Tests aggregated money-spent context generation: validates total_money_spent matches backend calculations
- Tests money-flow trend view: confirms period options, timeseries data, and chart payload generation for dashboard display
- Validates per-person money flows (who spent what, who owes what)

**Guest Registration Path (1 test case)**
- Validates invitation-based registration preserves room membership: guest → authenticated transition maintains user in room context

**UI/Template Updates**
- Added "Maybe register?" prompt to guest detail template with German translation included

### Business Impact

- **Risk Reduction**: Core financial flows (debt tracking, payment settlement) now have regression protection
- **User Experience Validation**: Guest onboarding prompts and registration pathway verified to display correctly
- **Cross-Functional Coverage**: Tests validate both backend calculations (money aggregation) and frontend rendering (templates, prompts)

### Required Follow-Up Actions

1. **Coverage Verification**: Validate that the new tests achieve 100% code coverage target for the targeted modules (debt views, room dashboard, transaction/money-flow views, account registration). Generate and attach coverage report to PR.

2. **Integration Check**: Run full test suite to ensure new tests pass and don't introduce fixture/dependency conflicts.

3. **Coverage Report Documentation**: If coverage target is not met, identify which code paths remain untested and determine if they need test coverage or should be excluded from coverage requirements.

4. **Edge Case Validation**: Verify tests cover:
   - Multiple debt scenarios (multi-party settlements, cascading debts)
   - Different guest registration flows (invitation vs. direct registration)
   - Localization for translation strings beyond German

### Test Quality Notes

- Tests leverage existing factories and fixtures for maintainability
- Context assertions validate both view logic and template data delivery
- Settlement test demonstrates state mutation patterns correctly (marking debt as settled, creating new transactions)
- Template content assertions use string matching; consider adding more specific HTML structure assertions if template changes are frequent

### CODE REVIEW

File: apps/transaction/tests/test_views/test_money_spent_views.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        expected_total = sum(transaction.value for transaction in child_transactions)
+
+        response = client.get(reverse("debt:money-spent-trend", kwargs={"room_slug": room.slug}))
+
+        assert response.status_code == http.HTTPStatus.OK
+        context = response.context_data
+
+        active_period = next(filter(lambda option: option["key"] == "last-activity", context["period_options"]))
+        assert active_period["active"] is True
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/transaction/tests/test_views/test_money_spent_views.py]: done

File: apps/transaction/tests/test_views/test_money_spent_views.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        assert total_entry["currency_sign"] == room.preferred_currency.sign
+
+        spent_per_person = list(context["money_spent_per_person_qs"])
+        owed_per_person = list(context["money_owed_per_person_qs"])
+
+        assert spent_per_person[0]["paid_by_name"] == user.name
+        assert spent_per_person[0]["total_spent_per_person"] == expected_total
+
+        assert owed_per_person[0]["paid_for__name"] == guest_user.name
+        assert owed_per_person[0]["total_owed_per_person"] == expected_total
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/transaction/tests/test_views/test_money_spent_views.py]: done

File: apps/room/tests/test_views/test_room_dashboard_view.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class TestRoomDashboardGuestOnboarding:
+    def test_guest_dashboard_shows_onboarding_prompts(self, client, room):
+        response = client.get(reverse("room:dashboard", kwargs={"room_slug": room.slug}))
+
+        assert response.status_code == http.HTTPStatus.OK
+        content = response.content.decode()
+        assert "Who are you?" in content
+        assert "Maybe register?" in content
+        assert "Create a free account" in content
```
[the remark this answers is not in the document]
Comment (christoph-teichmeister) [reply apps/room/tests/test_views/test_room_dashboard_view.py]: done
